common/autoware_universe_utils/test/src/geometry/test_geometry.cpp

```diff
@@ -1980,3 +1983,1053 @@ TEST(geometry, intersectPolygonRand)
+    std::printf(
+      "\tNot covered:\n\t\tBoost::geometry = %2.2f ms\n\t\tAlt = %2.2f ms\n",
+      ground_truth_not_covered_ns / 1e6, alt_not_covered_ns / 1e6);
+    std::printf(
+      "\tTotal:\n\t\tBoost::geometry = %2.2f ms\n\t\tAlt = %2.2f ms\n",
+      (ground_truth_not_covered_ns + ground_truth_covered_ns) / 1e6,
+      (alt_not_covered_ns + alt_covered_ns) / 1e6);
+  }
+}
+
+TEST(geometry, disjointRand)
+{
+  std::vector<autoware::universe_utils::Polygon2d> polygons;
+  constexpr auto polygons_nb = 500;
+  constexpr auto max_vertices = 10;
+  constexpr auto max_values = 1000;
+
+  autoware::universe_utils::StopWatch<std::chrono::nanoseconds, std::chrono::nanoseconds> sw;
+  for (auto vertices = 3UL; vertices < max_vertices; ++vertices) {
+    double ground_truth_disjoint_ns = 0.0;
+    double ground_truth_not_disjoint_ns = 0.0;
+    double alt_disjoint_ns = 0.0;
+    double alt_not_disjoint_ns = 0.0;
+    int disjoint_count = 0;
+
+    polygons.clear();
+    for (auto i = 0; i < polygons_nb; ++i) {
+      polygons.push_back(autoware::universe_utils::random_convex_polygon(vertices, max_values));
+    }
+    for (auto i = 0UL; i < polygons.size(); ++i) {
+      for (auto j = 0UL; j < polygons.size(); ++j) {
+        sw.tic();
+        const auto ground_truth = boost::geometry::disjoint(polygons[i], polygons[j]);
+        if (ground_truth) {
+          ++disjoint_count;
+          ground_truth_disjoint_ns += sw.toc();
+        } else {
+          ground_truth_not_disjoint_ns += sw.toc();
+        }
+
+        const auto alt_poly1 = autoware::universe_utils::alt::from_boost(polygons[i]);
+        const auto alt_poly2 = autoware::universe_utils::alt::from_boost(polygons[j]);
+        sw.tic();
+        const auto alt = autoware::universe_utils::disjoint(alt_poly1, alt_poly2);
+        if (alt) {
+          alt_disjoint_ns += sw.toc();
+        } else {
+          alt_not_disjoint_ns += sw.toc();
+        }
+
+        if (ground_truth != alt) {
+          std::cout << "Failed for the 2 polygons: ";
+          std::cout << boost::geometry::wkt(polygons[i]) << boost::geometry::wkt(polygons[j])
+                    << std::endl;
+        }
+        EXPECT_EQ(ground_truth, alt);
+      }
+    }
+    std::printf(
+      "polygons_nb = %d, vertices = %ld, %d / %d disjoint pairs\n", polygons_nb, vertices,
+      disjoint_count, polygons_nb * polygons_nb);
+    std::printf(
+      "\tDisjoint:\n\t\tBoost::geometry = %2.2f ms\n\t\tAlt = %2.2f ms\n",
+      ground_truth_disjoint_ns / 1e6, alt_disjoint_ns / 1e6);
+    std::printf(
+      "\tNot disjoint:\n\t\tBoost::geometry = %2.2f ms\n\t\tAlt = %2.2f ms\n",
+      ground_truth_not_disjoint_ns / 1e6, alt_not_disjoint_ns / 1e6);
+    std::printf(
+      "\tTotal:\n\t\tBoost::geometry = %2.2f ms\n\t\tAlt = %2.2f ms\n",
+      (ground_truth_not_disjoint_ns + ground_truth_disjoint_ns) / 1e6,
+      (alt_not_disjoint_ns + alt_disjoint_ns) / 1e6);
+  }
+}
+
+TEST(geometry, intersectsRand)
```

Comment:
This test (and probably the other random tests as well) should also use random non-convex polygons.
@mraditya01 is working on random polygon generation so maybe you can coordinate with him.